server/src/test/java/io/unitycatalog/server/persist/utils/TransactionManagerTest.java

```diff
@@ -0,0 +1,130 @@
+package io.unitycatalog.server.persist.utils;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.*;
+
+import io.unitycatalog.server.exception.BaseException;
+import io.unitycatalog.server.exception.ErrorCode;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.Transaction;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+public class TransactionManagerTest {
+
+  @Mock private SessionFactory sessionFactory;
+  @Mock private Session session;
+  @Mock private Transaction transaction;
+
+  @BeforeEach
+  public void setup() {
+    when(sessionFactory.openSession()).thenReturn(session);
+    when(session.beginTransaction()).thenReturn(transaction);
+  }
+
+  @Test
+  public void testSuccessfulTransaction() {
+    // Setup
+    String expectedResult = "Success";
+
+    // Execute
+    String result =
+        TransactionManager.executeWithTransaction(
+            sessionFactory, session -> expectedResult, "Error executing operation", false);
+
+    // Verify
+    assertThat(result).isEqualTo(expectedResult);
+    verify(transaction).commit();
+    verify(transaction, never()).rollback();
+    verify(session, never()).setDefaultReadOnly(true);
+  }
+
+  @Test
+  public void testReadOnlyTransaction() {
+    // Execute
+    TransactionManager.executeWithTransaction(
+        sessionFactory, session -> "Success", "Error executing operation", true);
```

Comment:
shouldnt you be testing by attempting to write in the readonly session?
is that not possible test using these mocks?